From 2498c77191086f04ac560bb3907da41e80bf87d8 Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@freescale.com>
Date: Fri, 16 Jan 2015 14:26:05 +0800
Subject: [PATCH] MLK-10107-2 usb: chipidea: add CONFIG_PM_RUNTIME for runtime
 pm API

For 3.19 kernel, it doesn't be needed due to CONFIG_PM_RUNTIME depends
on CONFIG_PM_SLEEP, but in 3.14, it still needs a standalone
CONFIG_PM_RUNTIME.

Signed-off-by: Peter Chen <peter.chen@freescale.com>
---
 drivers/usb/chipidea/ci_hdrc_imx.c | 2 ++
 drivers/usb/chipidea/core.c        | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/drivers/usb/chipidea/ci_hdrc_imx.c b/drivers/usb/chipidea/ci_hdrc_imx.c
index d5d9175..d5db00f 100644
--- a/drivers/usb/chipidea/ci_hdrc_imx.c
+++ b/drivers/usb/chipidea/ci_hdrc_imx.c
@@ -399,6 +399,7 @@ static int ci_hdrc_imx_resume(struct device *dev)
 }
 #endif /* CONFIG_PM_SLEEP */
 
+#ifdef CONFIG_PM_RUNTIME
 static int ci_hdrc_imx_runtime_suspend(struct device *dev)
 {
 	struct ci_hdrc_imx_data *data = dev_get_drvdata(dev);
@@ -422,6 +423,7 @@ static int ci_hdrc_imx_runtime_resume(struct device *dev)
 {
 	return imx_controller_resume(dev);
 }
+#endif /* CONFIG_PM_RUNTIME */
 
 #endif /* CONFIG_PM */
 
diff --git a/drivers/usb/chipidea/core.c b/drivers/usb/chipidea/core.c
index bac2829..e92019c 100644
--- a/drivers/usb/chipidea/core.c
+++ b/drivers/usb/chipidea/core.c
@@ -1044,6 +1044,7 @@ static int ci_resume(struct device *dev)
 }
 #endif /* CONFIG_PM_SLEEP */
 
+#ifdef CONFIG_PM_RUNTIME
 static int ci_runtime_suspend(struct device *dev)
 {
 	struct ci_hdrc *ci = dev_get_drvdata(dev);
@@ -1068,6 +1069,7 @@ static int ci_runtime_resume(struct device *dev)
 {
 	return ci_controller_resume(dev);
 }
+#endif /* CONFIG_PM_RUNTIME */
 
 #endif /* CONFIG_PM */
 static const struct dev_pm_ops ci_pm_ops = {
-- 
1.8.0

